feat(build): fluidattacks#1099 licensing builtin with reuse

rohaquinlop opened the pull request:
Fixes #1099 
- Create `testLicense` builtin with reuse
- Add job to dev pipeline

dsalaza4 commented on src/args/test-license/default.nix:
```diff
@@ -0,0 +1,16 @@
+  ...
+}:
+makeScript {
+  name = "test-license-for-${name}";
```
```suggestion
  name = "test-license";
```

dsalaza4 commented on src/evaluator/modules/test-license/default.nix:
```diff
@@ -0,0 +1,24 @@
+  config = {
+    outputs = {
+      "/testLicense" = lib.mkIf config.testLicense.enable (testLicense {
+        name = "builtin";
```
unnecessary as it is static?

dsalaza4 commented:
Please add documentation to the following section: https://makes.fluidattacks.com/api/builtins/test/

dsalaza4 commented on src/evaluator/modules/test-license/default.nix:
```diff
@@ -0,0 +1,22 @@
+}: {
+  options = {
+    testLicense = {
+      enable = lib.mkOption {
```
unnecessary? maybe just an empty map? `testLicense = {};`

dsalaza4 commented:
Use proper commit syntax. You can run `m . /lintGitCommitMsg`.

rohaquinlop commented:
> Please update your fork with upstream

Done.

dsalaza4 commented:
> > Please update your fork with upstream
> 
> Done.

Looks like you updated it using a merge commit instead of a rebase. Please make this PR to be only 1 commit ahead of upstream

rohaquinlop commented:
> > > Please update your fork with upstream
> > 
> > 
> > Done.
> 
> Looks like you updated it using a merge commit instead of a rebase. Please make this PR to be only 1 commit ahead of upstream

Done

dsalaza4 commented:
Please remove https://github.com/fluidattacks/makes/tree/main/makes/license:

- [ ] Makes job
- [ ] From GitHub actions jobs

dsalaza4 commented:
Usea the same PR message (title and body) as in your commit.

rohaquinlop commented:
> Please remove https://github.com/fluidattacks/makes/tree/main/makes/license:
> 
> * [ ]  Makes job
> * [ ]  From GitHub actions jobs

In dev and prod right?

dsalaza4 commented:
> 

Yes. As we're removing it, `/license` will not exist anymore. let's simply replace those github actions jobs that used it with the new `/testLicense`

rohaquinlop commented:
> > 
> 
> Yes. As we're removing it, `/license` will not exist anymore. let's simply replace those github actions jobs that used it with the new `/testLicense`

I already added the new one following that it's sorted alphabetically

dsalaza4 commented on .github/workflows/prod.yml:
```diff
@@ -357,27 +357,6 @@ jobs:
         env:
           CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
 
-  linux_license:
```
Let's make `/testLicense` run on prod too (replace these ones).

dsalaza4 commented:
Fix jobs:

- [ ] `/formatNix`
- [ ] `formatYaml`
- [ ] `lintGitMailMap` It is failing in remote for some reason?

rohaquinlop commented:
> 

Done.

dsalaza4 commented:
I'm not sure why but it looks like lintGitMailMap is asking for another entry when running on GitHub actions https://github.com/fluidattacks/makes/actions/runs/5259524687/jobs/9505213936?pr=1107

rohaquinlop commented:
> I'm not sure why but it looks like lintGitMailMap is asking for another entry when running on GitHub actions https://github.com/fluidattacks/makes/actions/runs/5259524687/jobs/9505213936?pr=1107

Hmmm guess it's because of my GitHub name? I'm gonna change it 😅

rohaquinlop commented:
@dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.

dsalaza4 commented:
> @dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.

I don't see any changes to the `.mailmap` file 🤔

dsalaza4 commented:
BTW, it runs every time you push a change, you should be able to see it here: https://github.com/fluidattacks/makes/pull/1107/checks

rohaquinlop commented:
> > @dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.
> 
> I don't see any changes to the `.mailmap` file 🤔

I mean, I changed my GitHub name, cause the name that was displaying the job was my GitHub name

dsalaza4 reviewed (approved):
LGTM!

Merging!
